packages/modules/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -1410,7 +1410,7 @@ export default class FulfillmentModuleService
         shippingOption.id
       )! // Garuantueed to exist since the validation above have been performed
 
-      if (shippingOption.type && !("id" in shippingOption.type)) {
+      if (shippingOption.type && typeof shippingOption.type === 'object' &&  !("id" in shippingOption.type)) {
```

Comment:
suggestion: you can use `isObject` when you need to do that and then we can simplify the condition in both places to `if (isObject(shippingOption.type) &&  !("id" in shippingOption.type))`